model/project.go

```diff
@@ -1431,38 +1457,66 @@ func (p *Project) FindAllVariants() []string {
 // for all variants of a project.
 func (p *Project) FindAllBuildVariantTasks() []BuildVariantTaskUnit {
 	tasksByName := map[string]ProjectTask{}
-	taskGroups := map[string]TaskGroup{}
 	for _, t := range p.Tasks {
 		tasksByName[t.Name] = t
 	}
-	for _, tg := range p.TaskGroups {
-		taskGroups[tg.Name] = tg
-	}
 	allBVTs := []BuildVariantTaskUnit{}
 	for _, b := range p.BuildVariants {
 		for _, t := range b.Tasks {
-			tasksToPopulate := []ProjectTask{}
+			t.Variant = b.Name
 			if t.IsGroup {
-				group, exists := taskGroups[t.Name]
-				if !exists {
-					continue
-				}
-				for _, groupTask := range group.Tasks {
-					tasksToPopulate = append(tasksToPopulate, tasksByName[groupTask])
-				}
+				allBVTs = append(allBVTs, p.tasksFromGroup(t)...)
 			} else {
-				tasksToPopulate = append(tasksToPopulate, tasksByName[t.Name])
-			}
-			for _, pTask := range tasksToPopulate {
-				t.Populate(pTask)
-				t.Variant = b.Name
+				t.Populate(tasksByName[t.Name])
 				allBVTs = append(allBVTs, t)
 			}
 		}
 	}
 	return allBVTs
 }
 
+// tasksFromGroup returns a slice of the task group's tasks.
+// Settings missing from the group task are populated from the task definition.
+func (p *Project) tasksFromGroup(bvTaskGroup BuildVariantTaskUnit) []BuildVariantTaskUnit {
+	tg := p.FindTaskGroup(bvTaskGroup.Name)
+	if tg == nil {
+		return nil
+	}
+
+	tasks := []BuildVariantTaskUnit{}
+	taskMap := map[string]ProjectTask{}
+	for _, projTask := range p.Tasks {
+		taskMap[projTask.Name] = projTask
+	}
+
+	for _, t := range tg.Tasks {
+		bvt := BuildVariantTaskUnit{
+			Name: t,
+			// IsGroup is not persisted, and indicates here that the
+			// task is a member of a task group.
+			IsGroup:          true,
+			Variant:          bvTaskGroup.Variant,
+			GroupName:        bvTaskGroup.Name,
+			Patchable:        bvTaskGroup.Patchable,
+			PatchOnly:        bvTaskGroup.PatchOnly,
+			Disable:          bvTaskGroup.Disable,
+			AllowForGitTag:   bvTaskGroup.AllowForGitTag,
+			GitTagOnly:       bvTaskGroup.GitTagOnly,
+			Priority:         bvTaskGroup.Priority,
+			DependsOn:        bvTaskGroup.DependsOn,
+			RunOn:            bvTaskGroup.RunOn,
+			ExecTimeoutSecs:  bvTaskGroup.ExecTimeoutSecs,
+			Stepback:         bvTaskGroup.Stepback,
+			Activate:         bvTaskGroup.Activate,
+			CommitQueueMerge: bvTaskGroup.CommitQueueMerge,
+		}
+		// Default to project task settings when unspecified
```

Comment:
nit: can we punctuate to match new style guidelines?

Reply:
I had a back and forth with Kim in the comments on the style guidelines where the conclusion was that proper punctuation was encouraged, but not required.